program_transformers/src/mpl_core_program/v1_asset.rs

```diff
@@ -536,3 +542,58 @@ fn convert_keys_to_snake_case(plugins_json: &mut Value) {
         _ => {}
     }
 }
+
+/// Updates the `asset_authority` for all assets that are part of a collection in a batch.
+/// This function performs a cursor-based paginated read and batch update.
+async fn update_group_asset_authorities<T: ConnectionTrait + TransactionTrait>(
+    conn: &T,
+    group_key: Vec<u8>,
+    authority: Vec<u8>,
+    slot: i64,
+) -> ProgramTransformerResult<()> {
+    let mut after = None;
+
+    let mut query = asset_grouping::Entity::find()
+        .select_only()
+        .column(asset_grouping::Column::AssetId)
+        .filter(asset_grouping::Column::GroupKey.eq(group_key.clone()))
```

Comment:
Group key is the word "collection" and group value is the bs58 encoded ID, so I think this needs to be:

```
let group_key = "collection".to_string();
let group_value = bs58::encode(id_vec.clone()).into_string();

let mut query = asset_grouping::Entity::find()
    .select_only()
    .column(asset_grouping::Column::AssetId)
    .filter(asset_grouping::Column::GroupKey.eq(group_key))
    .filter(asset_grouping::Column::GroupValue.eq(group_value))
    ...
```